src/expandableCalendar/agendaListNew.tsx

```diff
@@ -0,0 +1,259 @@
+import PropTypes from 'prop-types';
+
+import isUndefined from 'lodash/isUndefined';
+import debounce from 'lodash/debounce';
+import InfiniteList from '../infinite-list';
+
+import XDate from 'xdate';
+
+import React, {useCallback, useContext, useEffect, useMemo, useRef, useState} from 'react';
+import {
+  DefaultSectionT,
+  SectionListData,
+  LayoutChangeEvent,
+} from 'react-native';
+
+import {useDidUpdate} from '../hooks';
+import {getMoment} from '../momentResolver';
+import {isGTE, isToday} from '../dateutils';
+import {getDefaultLocale} from '../services';
+import {UpdateSources, todayString} from './commons';
+import styleConstructor from './style';
+import Context from './Context';
+import constants from "../commons/constants";
+import {parseDate} from "../interface";
+import {LayoutProvider} from "recyclerlistview/dist/reactnative/core/dependencies/LayoutProvider";
+import {AgendaListProps, AgendaSectionHeader} from "./agendaList";
+
+/**
+ * @description: AgendaList component that use InfiniteList to improve performance
+ * @note: Should be wrapped with 'CalendarProvider'
+ * @extends: SectionList
+ * @example: https://github.com/wix/react-native-calendars/blob/master/example/src/screens/expandableCalendar.js
+ */
+const AgendaList = (props: AgendaListProps) => {
```

Comment:
Rename the component AgendaListNew or InfiniteAgendaList

Reply:
Renamed to `InfiniteAgendaList`